### PR TITLE
fix(domain): validate domain and nameserver hostname shape before the API call

### PR DESCRIPTION
**Pull request by jpage-godaddy**
## Summary

[DEVEX-885](https://godaddy-corp.atlassian.net/browse/DEVEX-885) reports that `gddy domain` commands send raw, unvalidated input straight to the API:

```
gddy domain available "not a domain"       # → available: false
gddy domain available "https://test.com"   # → available: false
gddy domain available "test.com/page"      # → available: false
gddy domain available "test.com "          # trailing space preserved in output
gddy domain available $'test\x00evil.com'  # null byte truncates to "test"
```

Users get a confusing "not available" instead of a clear validation error.

- Add `validate_domain_name` in `domain/common.rs`, backed by `url::Host::parse` (already resolved in the dependency graph transitively via `reqwest`/`cli-engine` — no new crate added to the supply chain, just a new direct-dependency line) plus RFC 1035/1123 shape rules: rejects bare IP literals, requires ≥2 dot-separated LDH labels, ≤253 bytes total, and a non-numeric TLD.
- Applied to the `domain` argument in `available`, `get`, `quote`, and `nameservers set`.
- Every repeatable `--nameserver` host — in both `nameservers set` and `quote`'s own `--nameserver` flag (registration-profile nameservers) — goes through the same check via a shared `validate_nameserver_hosts`, so a bad host is reported as `--nameserver "..." is not a valid domain name`, not misattributed to the target domain (caught by Copilot review on the first pass — the initial version only covered `nameservers set`, missing `quote`'s own flag).
- Left `suggest`'s `query` alone — it's documented as accepting free-text keywords, not just domains.
- The validator itself is generic ("is not a valid domain name"); callers add their own context for non-domain arguments.

## Test plan

- [x] `cargo build`
- [x] `cargo test` (all passing, including new unit tests covering every case from the bug report plus IP literals, missing/numeric TLDs, hyphen-label edge cases, and the `--nameserver`-context wrapping)
- [x] `cargo clippy -- -D warnings`
- [x] `cargo fmt --check`
- [x] Manually ran all 5 repro commands from the ticket against `domain available`, confirmed `domain quote --nameserver <bad>` and `domain nameservers set --nameserver <bad>` are both now rejected, and confirmed `domain suggest <keyword>` still works unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: rust/src/domain/quote.rs [hunk 1]
```diff
@@ -228,12 +228,9 @@ pub(super) fn command() -> RuntimeCommandSpec {
...
         |ctx| async move {
-            let domain = ctx
-                .args
-                .get("domain")
-                .and_then(|v| v.as_str())
-                .unwrap_or("")
-                .to_owned();
+            let domain = validate_domain_name(
+                ctx.args.get("domain").and_then(|v| v.as_str()).unwrap_or(""),
+            )?;
```
[the remark this answers is not in the document]
Comment (jpage-godaddy) [reply rust/src/domain/quote.rs:236]: Good catch — fixed in 527536c: quote's own `--nameserver` flag (separate from `nameservers set`'s) now goes through the same validation. Factored the per-host check into a shared `validate_nameserver_hosts` in `common.rs` used by both commands, with unit tests covering the rejection.

**Review by qcai-godaddy — APPROVED** [review]
